Elmish.XamarinForms/ViewConverters.fs

```diff
@@ -135,8 +135,27 @@ module Converters =
 
     let identical (x: 'T) (y:'T) = System.Object.ReferenceEquals(x, y)
 
-    let canReuseChild (prevChild:ViewElement) (newChild:ViewElement) = 
-        (prevChild.TargetType = newChild.TargetType)
+    // NavigationPage can be reused only if the already existing pages don't change type (added/removed pages don't prevent reuse)
+    // E.g. If the first page switch from ContentPage to TabbedPage, the NavigationPage can't be reused; otherwise it can.
+    let canReuseNavigationPage (prevChild:ViewElement) (newChild:ViewElement) =
+        let prevPages = prevChild.TryGetAttribute<ViewElement[]>("Pages")
+        let newPages = newChild.TryGetAttribute<ViewElement[]>("Pages")
+
+        match prevPages, newPages with
+        | ValueSome prevPages, ValueSome newPages ->
+            let prevSeq = Seq.ofArray prevPages
+            let newSeq = Seq.ofArray newPages
```

Comment:
Array is automatically compatible with Seq, so you can remove these two conversions, thanks!

Also this is cleaner:

            (prevPages, newPages) ||> Seq.forall2 (fun prev curr -> prev.TargetType = curr.TargetType) 

In fact should this be

            (prevPages, newPages) ||> Seq.forall2 canReuseChild

just in case we add further special cases to `canReuseChild` for other types?

Reply:
Good point. Without `||>`, forall2 required explicit typing, so that's solved.

For canReuseChild, do you mean I should use a mutually recursive function to call canReuseChild inside canReuseNavigationPage?